src/cheri_insts.sail

```diff
@@ -419,8 +420,18 @@ function clause execute(CAndPerm(cd, cs1, rs2)) = {
   let perms = getCapPerms(cs1_val);
   let mask = truncate(rs2_val, cap_perms_width);
 
-  let inCap = clearTagIfSealed(cs1_val);
-  let newCap = setCapPerms(inCap, (perms & mask));
+  let newperms = perms & mask;
+
+  /*
+   * Sealed caps clear the tag unless the mask is all ones (~mask == zeros())
+   * or has the global permission clear.  This relies on the prop_null_noperms
+   * property to ensure that perm_global is a one-hot vector.
```

Comment:
```suggestion
   * CAndPerms on a sealed cap clears the tag unless the mask is all ones 
   * or has *only* the global permission clear.  Here `perm_global` is a 
   * vector of zeros except for the global permission bit. To construct this 
   * we rely on the fact that `null_cap` has all zero permissions.
```
Hopefully this is clearer. I removed the mention of prop_null_noperms as I thought it might just be confusing to people reading the arch doc. Good to have, though!